airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,486 @@
+    """basic stream for loading without sorting but with pagination
+    This logic can be used for a small data size when this data is loaded fast
+    """
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = self._parse_next_page_number(response)
+        if not next_page:
+            self._finished = True
+            return None
+        return next_page
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token)
+        params["page"] = next_page_token or 1
+        return params
+
+
+class FullRefreshBasicStream(IncrementalBasicUnsortedPageStream, ABC):
+    """"Basic stream for endpoints where there are not any created_at or updated_at fields"""
+
+    state_checkpoint_interval = None
+    cursor_field = []
+
+
+class IncrementalBasicSortedCursorStream(IncrementalBasicUnsortedStream, ABC):
+    """basic stream for loading sorting data with cursor based pagination"""
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token)
+        params.update({"sort_by": self.cursor_field, "sort_order": "desc", "limit": self.state_checkpoint_interval})
+
+        if next_page_token:
+            params["cursor"] = next_page_token
+        return params
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self.is_finished:
+            return None
+        return response.json().get("before_cursor")
+
+
+class IncrementalBasicSortedPageStream(IncrementalBasicUnsortedPageStream, ABC):
+    """basic stream for loading sorting data with normal pagination"""
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token, **kwargs)
+        if params:
+            params.update({"sort_by": self.cursor_field, "sort_order": "desc", "limit": self.state_checkpoint_interval})
+        return params
+
+
+class TicketComments(IncrementalBasicSortedPageStream):
+    """TicketComments stream: https://developer.zendesk.com/api-reference/ticketing/tickets/ticket_comments/
+    ZenDesk doesn't provide API for loading of all comments by one direct endpoints.
+    Thus at first we loads all updated tickets and after this tries to load all created/updated
+    comments per every ticket"""
+
+    response_list_name = "comments"
+    cursor_field = IncrementalBasicSortedPageStream.created_at_field
+
+    class Tickets(IncrementalBasicSortedPageStream):
```

Comment:
is there a need for this to be an inner class? we can make `include` an init param on the outer `tickets` stream and use it to craft query params. This way we don't duplicate the definition of the same entity in two different ways

Reply:
I see. Corrected